### PR TITLE
dooit: init at 1.0.1

### PR DESCRIPTION
**Pull request by wesleyjrz**
###### Description of changes

Just another nice todo program.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/dooit/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  # Required versions not available
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'textual = "^0.12.1"' 'textual = "*"' \
+      --replace 'tzlocal = "2.1"' 'tzlocal = ">2.1"'
+  '';
+
+  nativeBuildInputs = with python3.pkgs; [
+    poetry-core
+  ];
```
Comment (Kranzes) [inline pkgs/tools/misc/dooit/default.nix:27]:
```suggestion
  # Required versions not available
  pythonRelaxDeps = [
    "textual"
    "tzlocal"
  ];
  
  nativeBuildInputs = with python3.pkgs; [
    poetry-core
    pythonRelaxDepsHook
  ];
```

**Review by Kranzes — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -10253,6 +10253,8 @@ with pkgs;
 
   kb = callPackage ../tools/misc/kb { };
 
+  dooit = callPackage ../tools/misc/dooit { };
```
Comment (Kranzes) [inline pkgs/top-level/all-packages.nix:10256]: This file should be alphabetically ordered.

Comment (wesleyjrz) [reply pkgs/top-level/all-packages.nix:10256]: sure.

**Comment by Kranzes** [comment]
Please update to the latest version which was released yesterday. The new version fixes the crash which currently happens.

**Comment by Kranzes** [comment]
Result of `nixpkgs-review pr 219987` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>dooit</li>
    <li>dooit.dist</li>
  </ul>
</details>

**Pull request merged**